### PR TITLE
Tweaks for cran

### PR DESCRIPTION
**Pull request by mattsecrest**

### CODE REVIEW

**Comment by mattsecrest** [comment]
No response form Uwe yet let's see if this fixes things. 
[cran-submit-2.pdf](https://github.com/Genentech/psborrow2/files/14749335/cran-submit-2.pdf)

File: DESCRIPTION [hunk 1]
```diff
@@ -81,7 +81,8 @@ Suggests:
     table1,
     gt,
     gtsummary
-Additional_repositories: https://mc-stan.org/r-packages/
+Remotes:
+   stan-dev/cmdstanr
```
Comment (gravesti) [inline DESCRIPTION:85]: I'm sure the Additional_repositories way is what they need. Let's see if they can be more clear.... At least we could keep both.

File: DESCRIPTION [hunk 2]
```diff
@@ -27,9 +27,9 @@ Authors@R: c(
...
    data to supplement a randomized, controlled trial analysis in which
    external data are incorporated in a dynamic way (e.g., based on similarity
-   of outcomes); see Viele 2013 \doi{10.1002/pst.1589} for an overview.
+   of outcomes); see Viele 2013 <doi:10.1002/pst.1589> for an overview.
    This package implements the hierarchical commensurate prior approach to dynamic borrowing
-   as described in Hobbes 2011 \doi{10.1111/j.1541-0420.2011.01564.x}.
+   as described in Hobbes 2011 <doi:10.1111/j.1541-0420.2011.01564.x>.
    There are three main functionalities. First, 'psborrow2' provides a user-friendly 
    interface for applying dynamic borrowing on the study results handles the Markov Chain
    Monte Carlo sampling on behalf of the user. Second, 'psborrow2' provides a
```
Comment (gravesti) [inline DESCRIPTION:35]: From the CRAN policies

> Packages on which a CRAN package depends should be available from a mainstream repository: if any mentioned in ‘Suggests’ or ‘Enhances’ fields are not from such a repository, where to obtain them at a repository should be specified in an ‘Additional_repositories’ field of the DESCRIPTION file (as a comma-separated list of repository URLs) or for other means of access, described in the ‘Description’ field.

To me this reads that `Additional_repositories` is sufficient. My guess is they also want something about installing cmdstanr in `Description`

This package is designed to use the sampling functions from 'cmdstanr' which can be installed from <https://mc-stan.org/r-packages/>

**Comment by mattsecrest** [comment]
You have examples for unexported functions. Please either omit these
examples or export these functions.
Examples for unexported function
   c() in:
      add_covariates.Rd
      baseline_covariates.Rd
      check_data_matrix_has_columns.Rd
      covariance_matrix.Rd
      create_analysis_obj.Rd
      create_baseline_object.Rd
      create_data_matrix.Rd
      create_data_simulation.Rd
      create_simulation_obj.Rd
      enrollment_constant.Rd
      generate-BaselineObject-method.Rd
      generate-DataSimObject-method.Rd
      get_covariate_constraints.Rd
      mcmc_sample.Rd
      plot.Rd
      rename_draws_covariates.Rd
      set_enrollment.Rd
      sim_covariate_list.Rd
      sim_data_list.Rd

**Comment by mattsecrest** [comment]
\dontrun{} should only be used if the example really cannot be executed
(e.g. because of missing additional software, missing API keys, ...) by
the user. That's why wrapping examples in \dontrun{} adds the comment
("# Not run:") as a warning for the user. Does not seem necessary.
Please replace \dontrun with \donttest.
Please unwrap the examples if they are executable in < 5 sec, or replace
dontrun{} with \donttest{}.
-> mcmc_sample.Rd

**Comment by mattsecrest** [comment]
Per Daniel Sjoberg's suggestions just wrapping in @noRd and pushing back on `prior_gamma`, `prior_normal`, and `treatment_details`, which do not have commented out examples.

File: R/borrowing_details.R [hunk 3]
```diff
@@ -1,11 +1,18 @@
 #' Legacy function for specifying borrowing details
 #'
-#' Please use one of `hierarchical_commensurate_borrowing()`, `no_borrowing()`, or `full_borrowing()` instead.
+#' Please use one of `borrowing_hierarchical_commensurate()`, `borrowing_none()`, or `borrowing_full()` instead.
```
Comment (gravesti) [inline R/borrowing_details.R:3]:
```suggestion
#' Please use one of [borrowing_hierarchical_commensurate()], [borrowing_none()], or [borrowing_full()] instead.
```
I think making them links is even better so we get errors if the links don't work.

File: .github/workflows/R-CMD-check.yaml [hunk 4]
```diff
@@ -38,6 +38,7 @@ jobs:
         with:
           extra-packages: rcmdcheck, survival, flexsurv, testthat, usethis, vdiffr, tibble, xml2, knitr, rmarkdown, bayesplot, matrixcalc, WeightIt, MatchIt, BayesPPD, ggsurvfit, gbm, ggplot2, cobalt, table1, gt, gtsummary
           dependencies: '"hard"'
+          cache: false
```
Comment (gravesti) [inline .github/workflows/R-CMD-check.yaml:41]: Was this also to try to fix the windows issue?

File: R/generics.R [hunk 5]
```diff
@@ -68,6 +69,8 @@ setGeneric("show_guide", function(object) standardGeneric("show_guide"))
 #' @param object `MCMCSimulationResults` object
 #'
 #' @rdname get_results
+#' 
+#' @return data.frame with simulation results.
```
Comment (gravesti) [inline R/generics.R:73]:
```suggestion
#' @return A `data.frame` with simulation results.
```

**Review by gravesti — COMMENTED** [review]
Thanks for all those fixes @mattsecrest ! A few small ideas :)

**Pull request merged**